central/detection/lifecycle/manager_impl.go

```diff
@@ -133,6 +147,23 @@ func indicatorToBaselineKey(indicator *storage.ProcessIndicator) processBaseline
 	}
 }
 
+func (m *managerImpl) flushDeploymentQueue() {
+	defer centralMetrics.SetFunctionSegmentDuration(time.Now(), "CheckAndUpdateBaseline")
+
+	for {
+		// ObservationEnd is in the future so we have nothing to do at this time
+		head := m.deploymentQueue.Peek()
```

Comment:
because of the check on the observation period.  

head == nil || protoutils.After(head.ObservationEnd, types.TimestampNow())

If I pull and it is not nil, but the observation window has not been exceeded, then I would have to put it back.  I decided to use peek and put that logic here instead of in the observation_queue itself.  We could put that logic in the observation_queue.pull if you would rather it be there than here.